@here/harp-utils/lib/Functions.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+type chainFunction<T> = (...args: T[]) => T;
+
+/**
+ * Chains two functions for further assigning as one wrapped callback function
+ */
+export function chainCallbacks<T extends chainFunction<T>>(f1: any, f2: any): any {
```

Comment:
why generic T if not used at all ?

I know it's hard to write proper typings for this, at least shows the intent in typing albeit impl is not type-safe:
```
function chainCallbacks<T extends (this: unknown, ...args: any[]) => any>(
    f1: T | null | undefined,
    f2: T
): T {
    return function(this: any, ...args: any[]): ReturnType<T> {
        if (f1) {
            f1.apply(this, args);
        }
        return f2.apply(this, args);
    } as T;
}
```

Reply:
I understand it wrongly at the beginning. Thank you for the function code with typings